datafusion/core/src/dataframe.rs

```diff
@@ -302,6 +306,177 @@ impl DataFrame {
         ))
     }
 
+    /// Summary statistics for a DataFrame. Only summarizes numeric datatypes at the moment and
+    /// returns nulls for non numeric datatypes. Try in keep output similar to pandas
+    ///
+    /// ```
+    /// # use datafusion::prelude::*;
+    /// # use datafusion::error::Result;
+    /// # use arrow::util::pretty;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// let ctx = SessionContext::new();
+    /// let df = ctx.read_csv("tests/tpch-csv/customer.csv", CsvReadOptions::new()).await?;    
+    /// df.describe().await?;
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn describe(self) -> Result<()> {
+        Ok(print_batches(
```

Comment:
I think a nicer interface would be 

```rust
pub async fn describe(self) -> Result<Self> {
```

And then we don't need a separate `collect_describe` instead we could reuse the existing `collect()`

One way might be to return a new `DataFrame` from the created RecordBatch -- perhaps following the model of

https://docs.rs/datafusion/18.0.0/src/datafusion/execution/context.rs.html#719-730

Reply:
thanks , it's great suggestion.
fixed